### PR TITLE
Updating the build property from 'VisualStudioInstallDirectoryFromRegistry' to 'IsDev15BeforePreview4'

### PR DESCRIPTION
**Pull request by tannergooding**
FYI. @jasonmalinowski 

This cleans up the property name from https://github.com/dotnet/roslyn/pull/13086

### CODE REVIEW

File: build/Targets/Settings.targets [hunk 1]
```diff
@@ -133,18 +133,18 @@
        Basically, we check for a registry key that will only exist in legacy VS installs, and assume
        we are a willow based installation if our VSVersion is 15.0 and the registry key doesn't exist. -->
   <PropertyGroup>
-    <VisualStudioInstallDirectoryFromRegistry Condition="'$(OS)' == 'Windows_NT'">$(registry:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\VisualStudio\15.0@InstallDir)</VisualStudioInstallDirectoryFromRegistry>
-    <VisualStudioInstallDirectoryFromRegistry Condition="'$(OS)' == 'Windows_NT' AND '$(VisualStudioInstallDirectoryFromRegistry)' == ''">$(registry:HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\VisualStudio\15.0@InstallDir)</VisualStudioInstallDirectoryFromRegistry>
+    <IsDev15BeforePreview4 Condition="'$(OS)' == 'Windows_NT'">$(registry:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\VisualStudio\15.0@InstallDir)</IsDev15BeforePreview4>
+    <IsDev15BeforePreview4 Condition="'$(OS)' == 'Windows_NT' AND '$(IsDev15BeforePreview4)' == ''">$(registry:HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\VisualStudio\15.0@InstallDir)</IsDev15BeforePreview4>
```
Comment (jasonmalinowski) [inline build/Targets/Settings.targets]: My suggestion is to simply make this a Boolean-valued property, i.e.:

    <IsDev15BeforePreview4 Condition="$(registry:HKEY...)' != ''>true</IsDev15BeforePreview4>

Then you just have two cases that can set it to true, you don't have to worry about the ordering, etc.

Comment (tannergooding) [inline build/Targets/Settings.targets]: I will update.

Comment (tannergooding) [inline build/Targets/Settings.targets]: Updated.

**Comment by tannergooding** [comment]
Failing unit tests (under willow) are (FYI @genlu, @mavasani):
```
Microsoft.Build.Tasks.CodeAnalysis.UnitTests.dll                            FAILED 00:00:06.5796972
Roslyn.Services.Editor.CSharp.UnitTests.dll.5                               FAILED 00:00:54.8292115
Roslyn.InteractiveWindow.UnitTests.dll                                      FAILED 00:01:23.9383188
Roslyn.Services.Editor.UnitTests2.dll.2                                     FAILED 00:01:28.5480621
Roslyn.Services.Editor.VisualBasic.UnitTests.dll.5                          FAILED 00:01:32.5134392
Roslyn.Services.UnitTests.dll                                               FAILED 00:01:47.7283416
Roslyn.Services.Editor.UnitTests.dll                                        FAILED 00:01:51.5101008
Roslyn.Services.Editor.VisualBasic.UnitTests.dll.1                          FAILED 00:02:30.5284370
Roslyn.Services.Editor.VisualBasic.UnitTests.dll.2                          FAILED 00:02:35.7648719
Roslyn.VisualStudio.CSharp.UnitTests.dll                                    FAILED 00:02:48.7944688
Roslyn.VisualStudio.Services.UnitTests.dll.2                                FAILED 00:02:53.2333412
Roslyn.Services.Editor.VisualBasic.UnitTests.dll.3                          FAILED 00:02:56.1709641
Roslyn.Services.Editor.VisualBasic.UnitTests.dll.4                          FAILED 00:03:04.8548613
Roslyn.Services.Editor.CSharp.UnitTests.dll.1                               FAILED 00:03:43.6706682
Roslyn.Services.Editor.CSharp.UnitTests.dll.2                               FAILED 00:04:12.6469514
Roslyn.Services.Editor.UnitTests2.dll.1                                     FAILED 00:04:15.6256290
Roslyn.Services.Editor.CSharp.UnitTests.dll.4                               FAILED 00:04:20.7703500
Roslyn.Services.Editor.CSharp.UnitTests.dll.3                               FAILED 00:04:20.9402404
Roslyn.VisualStudio.Services.UnitTests.dll.1                                FAILED 00:05:35.4650072
```

**Comment by tannergooding** [comment]
FYI. @jaredpar, @dotnet/roslyn-infrastructure

**Comment by jasonmalinowski** [comment]
:+1:

**Pull request merged**